### PR TITLE
Add dot1{ad,q} traffic modes

### PR DESCRIPTION
**Pull request by sirspinach**
Dot1ADMode will not run without Scapy 2.3.3+ (needs dot1AD layer).

### CODE REVIEW

**Comment by sirspinach** [comment]
I think I'll also add encap/decap options so we can easily deal with VLAN-configured switches.

**Comment by melvinw** [comment]
Good idea

**Comment by melvinw** [comment]
One more request. Could you update the readme with a note about needing/how to get scapy 2.3.3+?

**Comment by chris3torek** [comment]
This is where / why I had to do `pip install git+https://github.com/secdev/scapy`, right? I don't see it in here, and maybe this should go into both this repo's .travis.yml and the NetSys/bess .travis.yml (it looks like it's already in bess/env/packages.yml though, and clearly I don't understand how these all interact...)

**Comment by melvinw** [comment]
This repo isn't configured to use travis. It would be nice to set it up to catch when BESS changes break things though, but that's beyond the scope of this PR and requires some more thought on my part-- I don't really know what sort of tests are meaningful and can also be carried out on travis.

**Comment by chris3torek** [comment]
Interesting. It _has_ a `.travis.yml` file ... (plus I've been musing about how to redo the vhost DPDK overlay stuff and move it here, but have not come to any conclusion at all)

**Comment by chris3torek** [comment]
Oops, I'm blind, it's a different repo (nefeli/bess-internal) that has its own .travis.yml

File: generator/modes/dot1ad.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            pkt_templates = [_build_pkt(spec, spec.pkt_size)]
+
+        # Setup tx pipeline
+        return Pipeline([
+            Source(),
+            Rewrite(templates=pkt_templates),
+            IPChecksum(),
+            VLANPush(),
+            VLANPush(),
+            RandomUpdate(fields=[{'offset': 14,
```
Comment (sirspinach) [inline generator/modes/dot1ad.py]: fixed

**Comment by melvinw** [comment]
Thanks!

**Pull request merged**